src/tests/test_database.py

```
@@ -97,17 +119,464 @@ def test_add_task_with_partial_fields(self):
+    @classmethod
+    def setUpClass(cls):
+        """Set up test class."""
+        if os.path.exists(cls.TEST_DB_NAME):
+            try:
+                os.remove(cls.TEST_DB_NAME)
+            except PermissionError:
+                time.sleep(1.0)  # Increased delay for Windows
+                try:
+                    os.remove(cls.TEST_DB_NAME)
+                except PermissionError:
+                    pass
```

Ruff (0.8.0)

Use `contextlib.suppress(PermissionError)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(PermissionError)`

(SIM105)